src/libraries/Native/Unix/System.Security.Cryptography.Native/android_security.c

```diff
@@ -0,0 +1,196 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_err.h"
+#include "pal_types.h"
+#include "pal_utilities.h"
+#include "pal_safecrt.h"
+
+#include "android_security.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <jni.h>
+#include <android/log.h>
+
+static JavaVM *gJvm;
+
+PALEXPORT JNIEXPORT jint JNICALL
+JNI_OnLoad(JavaVM *vm, void *reserved)
+{
+    (void)reserved;
+    __android_log_write(ANDROID_LOG_INFO, "DOTNET", "JNI_OnLoad in android_security.c");
+    gJvm = vm;
+    return JNI_VERSION_1_6;
+}
+
+static jobject ToGRef(JNIEnv *env, jobject lref)
+{
+    if (lref == 0)
+        return 0;
+    jobject gref = (*env)->NewGlobalRef(env, lref);
+    (*env)->DeleteLocalRef(env, lref);
+    return gref;
+}
+
+static JNIEnv* GetJniEnv()
+{
+    JNIEnv *env;
+    (*gJvm)->GetEnv(gJvm, (void**)&env, JNI_VERSION_1_6);
+    if (env)
+        return env;
+    (*gJvm)->AttachCurrentThread(gJvm, &env, NULL);
+    return env;
+}
+
+int32_t CryptoNative_GetRandomBytes(uint8_t* buff, int32_t len)
+{
+    JNIEnv* jniEnv = GetJniEnv();
+
+    jclass randClass = (*jniEnv)->FindClass (jniEnv, "java/security/SecureRandom");
```

Comment:
You should *cache* the `randClass`, `randCtor`, and `randNextBytes` values as global values.  `JNIEnv::FindClass()` & co. are not "fast", and it would be best to avoid these calls on every invocation.

Note that `randClass` will need to be held as a GREF; `ToGRef()` already exists.